### PR TITLE
chore(deps): bump CDK to 2.114.1

### PR DESCRIPTION
**Pull request by marofke**
## Problem
We needed to update the version of CDK used by RFDK.

## Solution
Ran `npm-upgrade` for all relevant `package.json` files, bumping everything that made sense. I had to make a few minor tweaks to tests in order for them to build properly, and changed some deprecated function calls to get rid of warnings. I also took this opportunity to clean up the dependabot ignorelist for CDK-related packages. Also updated the python examples' CDK version since that was behind since the last upgrade.

## Testing
- `yarn run build`
- Deployed the `All-In-Infrastructure-Basic` stack in my AWS account, confirmed it spun up and tore down successfully

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by jericht — DISMISSED** [review]
LGTM!

**Review by jusiskin — DISMISSED** [review]
Great work on this large migration @marofke!

File: examples/deadline/Local-Zone/python/package/lib/config.py [hunk 1]
```diff
@@ -37,12 +37,12 @@ def __init__(self):
         # The version of Deadline to use on the render farm. Leave as None for the latest release or specify a version
         # to pin to. Some examples of pinned version values are "10", "10.1", or "10.1.16"
         # The default value of 10.1.19 is used, to match the worker AMI ID provided below
```
Comment (jericht) [inline examples/deadline/Local-Zone/python/package/lib/config.py]: nit: comment version mismatch
```suggestion
        # The default value of 10.3.1 is used, to match the worker AMI ID provided below
```

File: examples/deadline/Local-Zone/ts/bin/config.ts [hunk 2]
```diff
@@ -38,14 +38,14 @@ class AppConfig {
    * "10.1.16"
    * @default 10.1.20 is used, to match the worker AMI ID provided below
```
Comment (jericht) [inline examples/deadline/Local-Zone/ts/bin/config.ts]: Default needs updating
```suggestion
   * @default 10.3.1 is used, to match the worker AMI ID provided below
```

**Review by jericht — CHANGES_REQUESTED** [review]

**Review by jericht — APPROVED** [review]

**Review by jusiskin — APPROVED** [review]

**Pull request merged**